### PR TITLE
Add a JMH benchmark for `ThriftMessageClassFinder`

### PR DESCRIPTION
**Pull request by moonchanyong**
Motivation:

add benchmarking result in line of PR: #2055 

Modification:

* add benchmarking result about between using Thread.getStackTrace, Stackwalker, and Stackwalker with RETAIN_CLASS_REFERENCE option

Result:

* Throughput of Stackwalker with RETAIN_CLASS_REFERENCE option is better than others 
* but is recoded high error

### CODE REVIEW

**Comment by trustin** [comment]
Thanks a lot! Could you paste the result?

**Comment by trustin** [comment]
I think you must pass the value returned by .get().

**Comment by trustin** [comment]
The result after adding `.get()` - 5 times faster!

```
Benchmark                                          Mode  Cnt       Score       Error  Units
StackTracingBenchmark.defaultStackTrace           thrpt    5  104310.565 ±  5994.780  ops/s
StackTracingBenchmark.usingStackWalker            thrpt    5  269399.892 ±  6504.989  ops/s
StackTracingBenchmark.usingStackWalkerWithOption  thrpt    5  538596.277 ± 96659.058  ops/s
```

**Comment by moonchanyong** [comment]
I forgot to add get method after remove proxy method for calling on TBase implemented class.
I pushed again

File: benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        });
+
+        stackWalkingThriftMessageClassFinder = (Supplier<Class<?>>) stackWalkerConstructor.newInstance();
+
+        System.setSecurityManager(SM);
+    }
+
+    @TearDown
+    public void stopServer() {
+    }
```
Comment (trustin) [inline benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java]: Perhaps we do not need this method?

File: benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    private Supplier<Class<?>> defaultThriftMessageClassFinderfault;
+    private Supplier<Class<?>> stackWalkingThriftMessageClassFinder;
+    private Supplier<Class<?>> stackWalkingFinderWithRetainOption;
+
+    @Setup
+    public void makeFinder() throws Exception {
+        final Class<?> defaultThriftMessageClassFinderClass =
+                Class.forName("com.linecorp.armeria.common.thrift.text.DefaultThriftMessageClassFinder");
+        final Class<?> stackWalkingThriftMessageClassFinderClass =
+                Class.forName("com.linecorp.armeria.common.thrift.text.StackWalkingThriftMessageClassFinder");
```
Comment (trustin) [inline benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java]: Perhaps we do not use reflection if we put this benchmark class in the same package?

File: benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+import com.linecorp.armeria.thrift.services.HelloService.hello_args;
+
+/**
+ * Compare performance of Thread.getStackTrace vs StackWalker(reflected) vs StackWalker(reflected) with Option
```
Comment (ikhoon) [inline benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java]: nit: `{@link Thread.getStackTrace}` and others?

File: benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java (hunk 2 above)
Comment (moonchanyong) [reply benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java]: I didn't think that. thanks for your advice.

File: benchmarks/src/jmh/java/com/linecorp/armeria/common/thrift/text/StackTracingBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+import com.linecorp.armeria.thrift.services.HelloService.hello_args;
+
+/**
+ * Compare performance of {@link Thread#getStackTrace} vs StackWalker vs StackWalker with Option
```
Comment (trustin) [inline benchmarks/src/jmh/java/com/linecorp/armeria/common/thrift/text/StackTracingBenchmark.java]:
```
Compare the performance between {@link Thread#getStackTrace()}, {@code StackWalker}
and {@code StackWalker} with the {@code RETAIN_CLASS_REFERENCE} option when finding
the current Thrift message class.
```

**Pull request merged**